roscala/src/main/scala/coop/rchain/rosette/prim/tuple.scala

```diff
@@ -153,6 +153,77 @@ object tuple {
     }
   }
 
+  /**
+    * Define the tuple-head primitive.
+    * This returns the first element of a Tuple
+    * e.g. (tuple-head [1 2 3 4 5 6]) ==> 1
+    */
+  object tplHead extends Prim {
+    override val name: String = "tuple-head"
+    override val minArgs: Int = 1
+    override val maxArgs: Int = 1
+
+    @checkTypeMismatch[Tuple] // Only arg must be a Tuple
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Ob] = {
+      val elem = ctxt.argvec.elem
+
+      checkTuple(0, elem).map { t => // Ensure arg0 is a Tuple
+
+        t.nth(0) match {
+          case Some(v: Ob) => v
+          case None        => Ob.NIV
```

Comment:
Let's return `NIL` here.

Reply:
What's the syntax for that, or where is NIL defined? The compiler says "not found: value NIL".

Reply:
I tried "Nil.asInstanceOf[Ob]", but the test refuses to cast Nil to an Ob at runtime. Suggestions?

Reply:
There should be `Tuple.NIL`. That doesn't work?

Reply:
Done